File: tmp-mei/fgen4255027576711118260/base/Debug_0.java
package base;
public interface Debug_0 extends base.Sealed_0{
Debug_0 $self = new Debug_0Impl();
base.Void_0 println$imm(Object x_m$);

rt.Str identify$imm(Object x_m$);

Object $hash$imm(Object x_m$);
static Object $hash$imm$fun(Object x_m$, base.Debug_0 $this) {
  return (switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (Object) null;
}})
;
}

static base.Void_0 println$imm$fun(Object x_m$, base.Debug_0 $this) {
  return ((base.Void_0)(switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (base.Void_0) null;
}})
);
}

static rt.Str identify$imm$fun(Object x_m$, base.Debug_0 $this) {
  return ((rt.Str)(switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (rt.Str) null;
}})
);
}
}

File: tmp-mei/fgenjs5030319945734616766/base/Debug_0.js
import { base$$Sealed_0 } from "../base/Sealed_0.js";
import * as rt from "../rt/main.js";

export class base$$Debug_0 extends base$$Sealed_0 {
  static $self = new base$$Debug_0();

  println$imm(x_m$) {
  return base$$Debug_0.println$imm$fun(x_m$, this);
}


identify$imm(x_m$) {
  return base$$Debug_0.identify$imm$fun(x_m$, this);
}


$hash$imm(x_m$) {
  return base$$Debug_0.$hash$imm$fun(x_m$, this);
}

  static $hash$imm$fun(x_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

    static println$imm$fun(x_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

    static identify$imm$fun(x_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

}
